### ISSUE TITLE
Conversion stopped working after a year

### ISSUE THREAD

**Issue by sirlanslot** [documents4j/documents4j#83]
I have 2 servers running the same code set using the local converter for ms word, the process successfully converted docx files to,pdf and now I’m receiving a ConverterAccessException message indicating the converter seems to have shutdown.  The process starts the word application but fails to convert the pdf, this does this on every document it tries to convert

Word is being started as the launched user in the dcom configuration.

The process has been working with java8 and word 2016 and word 2019 without any issues until now.  I’ve tried reinstalling word, restarted the server, changed the configuration on the service user and nothing seems to get the process to successfully convert a document to pdf. 

Any ideas on what could be the issue for this sudden failure. Basically what 8m seeing, word starts then the converter shutsdown preventing the conversion process

**Comment by raphw** [comment]
Have you tried to run with a "real" user and not with the service user to see if it is working then? Typically, this is where the problem lies. Did you update Windows when it stopped working? If it works with a regular user, it is probably related to the service user's configuration and maybe an update changed the way non-headless applications work on it. You typically need to make sure several folders exist and maybe that has changed for your particular Windows version.

**Comment by sirlanslot** [comment]
no updates occurred in the last 30 days. As I indicated, it worked up until last week, then just stopped. As for a real user, not sure what the ask is..I have a local user that runs the the application which in turns calls the word process.  I also can see the word process starting on each iteration, the is operational check now always returns false with the message I mentioned previously. Which folders are you referring to, I’m only aware of the base folder for the api and the desktop folder if running as a service.  Are there others that I need to check..thanks

**Comment by raphw** [comment]
I meant the difference between a logon user and a service user, the service user can be a bit tricky. If the converter seems to be shut down, there is a chance that Word tries to show an error message upon start up, for example a recovery mode window that is not possible in headless mode. Did you try to open word directly with the user being used for running the server?

**Comment by sirlanslot** [comment]
I was able to open word manually,  no errors on open, no wizards or confirmation dialogs. Is there a way I can display word during the process to see if there’s an error or is there a debug mode where I can capture what word is doing?   I also created a new user made it an admin and received the same message.

**Comment by raphw** [comment]
Yes, take the VBS scripts (https://github.com/documents4j/documents4j/tree/master/documents4j-transformer-msoffice/documents4j-transformer-msoffice-word/src/main/resources) and run them directly on the command line. Run the start script and then the assert script to see why it cannot connect to Word. It might give you a pop up with an error. Otherwise, look at the VBS API to extract further errors.

**Comment by sirlanslot** [comment]
The assert vbs  script produced no errors. I also did not see any errors when I opened word with the start script the visible property to true. I’ll need to look into another solution since this one is a difficult one to debug. My next test would be to spin up a new server. Hopefully this one works.

**Comment by sirlanslot** [comment]
So the behavior that I’m seeing is the following. Word start scripts starts a word process but the system is not able to connect to this process so it thinks it has shutdown and then fails the conversion.  If I leave the word process running and rerun the word start script, another instance of word is created.  If I run word shutdown, it terminates the last one started.

**Comment by sirlanslot** [comment]
What I’m finding is if the application starts word via the start script, when I run the word_assert vbs script with edited info, I receive a message that activex component cant be created, with error number 429. If I run the start and the assert manually ie outside the application I receive no error message and the assert returns 0. So there is something preventing the word from connecting to the convertor. Any thoughts.

**Comment by raphw** [comment]
Never heard of that error; did you try to fix the ActiveX issue? (https://windowsreport.com/activex-error-429-fix/) If it helps, why not, no idea why Word needs it.

**Comment by whammoed** [comment]
Getting same issue still.  The application is running as a service.  I can see the WINWORD.EXE was created in the task manager but the application seems to fail on:
 Set wordApplication = GetObject(, "Word.Application")
 Running Word_Assert from the command line confirms as I get 429 ActiveX component can't create object
If I run Word_Start.vbs from the command line it is successful and then so is Word_Assert.vbs (I now have 2 WINWORD.EXE running)
At this point my application still will not work and still gives "com.documents4j.throwables.ConversionInputException: The input file seems to be corrupt"
Will probably see if I can get office365 installed instead.  Works on my dev machine.

**Comment by whammoed** [comment]
Update: currently getting the activation window popping up when I open Word manually so I will test again once resolved.  This wasn't happening when I was testing previously but could be the issue.

**Comment by whammoed** [comment]
Update: Unfortunately still getting same behavior after activation.  Works with Office 2016 but does not work with Office 2019.

**Comment by whammoed** [comment]
Update:  working after creating folder C:\Windows\System32\config\systemprofile\Desktop
This is a 64bit machine with the 64bit version of Office 2019 so I don't know why this folder is needed when it was not for Office 2016.  Very strange.

**Comment by raphw** [comment]
You are likely running Office in 32 bit mode.

**Comment by whammoed** [comment]
Says 64 bit on the about screen:

![office 64 bit](https://github.com/user-attachments/assets/586103c3-2374-424d-bb9e-022191287af2)
